server/notification-providers/send-grid.js

```diff
@@ -0,0 +1,65 @@
+const NotificationProvider = require("./notification-provider");
+const axios = require("axios");
+
+class SendGrid extends NotificationProvider {
+  name = "SendGrid";
+
+  /**
+   * @inheritdoc
+   */
+  async send(notification, msg, monitorJSON = null, heartbeatJSON = null) {
+    const okMsg = "Sent Successfully.";
+
+    try {
+      let config = {
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${notification.sendgridApiKey}`,
+        },
+      };
+
+      let personalizations = {
+        to: [{ email: notification.sendgridToEmail }],
+      };
+
+      // Add CC recipients if provided
+      if (notification.sendgridCcEmail) {
+        personalizations.cc = notification.sendgridCcEmail
+          .split(",")
+          .map((email) => ({ email: email.trim() }));
+      }
+
+      // Add BCC recipients if provided
+      if (notification.sendgridBccEmail) {
+        personalizations.bcc = notification.sendgridBccEmail
+          .split(",")
+          .map((email) => ({ email: email.trim() }));
+      }
+
+      let data = {
+        personalizations: [personalizations],
+        from: { email: notification.sendgridFromEmail },
+        subject:
+          notification.sendgridSubject ||
```

Comment:
Please make the default more clear to our users by setting it like this instead.

(Read: leave this code, but also add a default in the frontend)

Here is an example how you do this:
https://github.com/louislam/uptime-kuma/blob/85dfe1f5d17dad5e035eb3620b8f516b2bcf79cd/src/components/notifications/TechulusPush.vue#L66-L69

Reply:
done [5f96e67](https://github.com/louislam/uptime-kuma/pull/5205/commits/5f96e6700ed45125cedb3d96051eba773354a741)